src/WebJobs.Script/Config/ActiveHostConfigurationSource.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    public class ActiveHostConfigurationSource : IConfigurationSource
+    {
+        public ActiveHostConfigurationSource(IServiceProvider serviceProvider)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        public IServiceProvider ServiceProvider { get; set; }
+
+        public IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            return new ActiveHostConfigurationProvider(this);
```

Comment:
With the current implementation, passing the service provider would be sufficient.